tests/test_cbv_base.py

```diff
@@ -1,88 +1,137 @@
 
-    response_2 = client.get("/classvar")
-    assert response_2.status_code == 200
-    assert response_2.content == b"false"
+class TestCbv(AbstractTest):
+    WANTED_ATTR ="cy"
+    INPUTTED = 2
+    BUILTIN = 1
+
+    @pytest.fixture
+    def cbv_resource(self):
+        class CBV(Resource):
+            def __init__(self, user_input: int = 1):
+                super().__init__()
+                self.builtin_value = TestCbv.BUILTIN
+                self.configured_value = user_input
 
+            @set_responses(int)
+            def post(self, user_given: int) -> int:
+                print(user_given)
+                return user_given + self.builtin_value + self.configured_value
 
-def test_arg_in_path() -> None:
-    class TestCBV(Resource):
-        @set_responses(str)
-        def get(self, item_id: str) -> str:
-            return item_id
+            @set_responses(bool)
+            def get(self) -> bool:
+                return hasattr(self, TestCbv.WANTED_ATTR)
 
-    app = FastAPI()
-    api = Api(app)
+        return CBV
 
-    test_cbv_resource = TestCBV()
-    api.add_resource(test_cbv_resource, "/{item_id}")
+    def setup_method(self):
+        self.posted_value = 2
 
-    assert TestClient(app).get("/test").json() == "test"
+    @pytest.fixture(autouse=True)
+    def add_resource(self, api: Api, cbv_resource: object):
+        cbv = cbv_resource(self.INPUTTED)
+        api.add_resource(cbv, "/", "/classvar")
 
+    def test_post_params(self, client: TestClient):
+        response_post = client.post("/", params={
+                                    "user_given": self.posted_value}, json={})
+        summed = self.posted_value + self.BUILTIN + self.INPUTTED
+        assert response_post.status_code == status.HTTP_200_OK
+        assert response_post.json() == summed
 
-def test_multiple_routes() -> None:
-    class RootHandler(Resource):
-        def get(self, item_path: str = None) -> Union[List[Any], Dict[str, str]]:
-            if item_path:
-                return {"item_path": item_path}
-            return []
+    def test_get_classvar(self, client: TestClient):
+        response_2 = client.get("/classvar")
+        assert response_2.status_code == status.HTTP_200_OK
+        assert response_2.json() is False
 
-    app = FastAPI()
-    api = Api(app)
 
-    root_handler_resource = RootHandler()
-    api.add_resource(root_handler_resource, "/items/?", "/items/{item_path:path}")
+class TestArgsInPath(AbstractTest):
+    ITEM = "test"
 
-    client = TestClient(app)
+    @pytest.fixture
+    def cbv_resource(self):
+        class TestCBV(Resource):
+            @set_responses(str)
+            def get(self, item_id: str) -> str:
+                return item_id
 
-    assert client.get("/items/1").json() == {"item_path": "1"}
-    assert client.get("/items").json() == []
+        return TestCBV
 
+    @staticmethod
+    @pytest.fixture(autouse=True)
+    def add_resource(cbv_resource: object, api: Api):
```

Comment:
I was sooooo confused from this
What had happened here? this is not commonly used and therefore not needed a separated fixture 
rest easy to merge this fixture with the one setting up application as they are the same